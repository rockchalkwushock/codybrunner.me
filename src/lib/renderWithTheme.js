import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from './theme'

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
