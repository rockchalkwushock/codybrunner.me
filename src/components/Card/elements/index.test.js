import React from 'react'

import renderWithTheme from '../../../lib/renderWithTheme'
import { Card, CardDate, CardExcerpt, CardTitle } from './'

describe('Component: <Card />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<Card draft={false} />)
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper.children().props()).toHaveProperty('draft', false)
    expect(wrapper).toHaveStyleRule('background-color', '#7a9eb1')
    expect(wrapper).toHaveStyleRule('border', 'none')
    expect(wrapper).toHaveStyleRule('border-radius', '10px')
    expect(wrapper).toHaveStyleRule(
      'box-shadow',
      '1px 1px 5px 1px rgba(0,0,0,0.7)'
    )
    expect(wrapper).toHaveStyleRule('padding', '1em')
    expect(wrapper).toHaveStyleRule('margin', '0 auto')
    expect(wrapper).toHaveStyleRule('transition', 'all 0.3s ease-in-out')
    expect(wrapper).toHaveStyleRule('box-shadow', 'unset', {
      modifier: ':hover'
    })
    expect(wrapper).toHaveStyleRule('width', '60%', {
      media: '(min-width:736px)'
    })
  })
  test('should have different style with prop', () => {
    wrapper = renderWithTheme(<Card draft />)
    expect(wrapper.children().props()).toHaveProperty('draft', true)
    expect(wrapper).toHaveStyleRule('border', '6px dashed red')
  })
})

describe('Component: <CardDate />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<CardDate />)
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('padding-bottom', '0.5em')
  })
})

describe('Component: <CardExcerpt />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<CardExcerpt />)
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have styles', () => {
    expect(wrapper).toHaveStyleRule('text-align', 'justify')
  })
})

describe('Component: <CardTitle />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = renderWithTheme(<CardTitle />)
  })
  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
