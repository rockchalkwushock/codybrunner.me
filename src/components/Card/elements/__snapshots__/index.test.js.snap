// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Card /> should render correctly 1`] = `
.c0 {
  background-color: #7a9eb1;
  border: none;
  border-radius: 10px;
  box-shadow: 1px 1px 5px 1px rgba(0,0,0,0.7);
  padding: 1em;
  margin: 0 auto;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c0:hover {
  box-shadow: unset;
}

@media (min-width:736px) {
  .c0 {
    width: 60%;
  }
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <Card
    draft={false}
  >
    <div
      className="c0"
    />
  </Card>
</ThemeProvider>
`;

exports[`Component: <CardDate /> should render correctly 1`] = `
.c0 {
  padding-bottom: 0.5em;
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <Date>
    <h4
      className="c0"
    />
  </Date>
</ThemeProvider>
`;

exports[`Component: <CardExcerpt /> should render correctly 1`] = `
.c0 {
  text-align: justify;
}

<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <Excerpt>
    <p
      className="c0"
    />
  </Excerpt>
</ThemeProvider>
`;

exports[`Component: <CardTitle /> should render correctly 1`] = `
<ThemeProvider
  theme={
    Object {
      "post": Object {
        "bg": "#f9f8eb",
        "ftClr": "#000",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1.1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#7a9eb1",
        "linkWt": "600",
        "lnHt": "1.618",
        "ltSp": "1px",
      },
      "screen": Object {
        "large": "736px",
      },
      "site": Object {
        "bg": "#7a9eb1",
        "borderRadius": "10px",
        "fg": "#ffe1b6",
        "ftClr": "#143a52",
        "ftFm": "\\"Raleway\\", sans-serif",
        "ftSz": "1rem",
        "ftWt": "400",
        "linkClr": "#143a52",
        "linkHvClr": "#ffe1b6",
        "lnHt": "1.4",
        "ltSp": "0px",
        "shadow": "1px 1px 5px 1px rgba(0, 0, 0, 0.7)",
        "transition": "all 0.3s ease-in-out",
      },
    }
  }
>
  <Title>
    <h2
      className=""
    />
  </Title>
</ThemeProvider>
`;
